examples/test_json.cc

```diff
@@ -0,0 +1,116 @@
+/**
+ * file   test_nl.cc
+ *
+ * @author Felix Musil <felix.musil@epfl.ch>
+ *
+ * @date   2018
+ *
+ * @brief Example for Neighbour list
```

Comment:
would you prefer playground rather than test because that is what it is

Reply:
Yeah, something like `sandbox.cc`; ideally this wouldn't get pushed to master.

Reply:
The purpose of this file is to keep track of the development of the code so that when I want to build a new functionality I can start from there (avoiding writing over and over again the same set up code) to see if it compiles and makes at least some sense.
I agree with you that it is meant to be deleted at some point since the development effort will mostly have converged.

Reply:
I tend to agree with Felix regarding the functionality, but with Max for the design.
My ideal version would be
1. Example folder, where we have 'c++' examples on how to use and iterate - kind of like this folder started
2. A test/sandbox/temp/playground folder which is fair game

Ideally the sandbox-programs would migrate to examples once functionality is added. And I would like to keep the examples simple.
I once worked with a finite element library called 'deal.ii' and I think I mentioned that before. They have a very nice way of combining examples with a tutorial. [Link](https://www.dealii.org/current/doxygen/deal.II/Tutorial.html). So when you want to use their library, you do the first couple of tutorial, get the hang of it and then use one of the more complex examples, which is close to your problem and modify it.

Reply:
I like this idea. If we make clear that the sandbox folder contains unstable examples, it should be fine. If we want to make it super professional in the future, we only keep this folder in a main development branch and ignore it when merging, but I don't think this is necessary for a science library.

Reply:
Are you saying that the code doesn't need to be professional because it's a "science library"? On the contrary, because so many people rely on scientific libraries to get accurate results for their research, that's all the more reason to keep the code clean, stable, and professional. And that applies to the examples as well; eventually we'll want to convert all the sandbox code into usable, predictable, well-documented examples (and the library that Markus linked is no different: All their examples serve a purpose and are extensively documented!).

Reply:
So do we want to make a `sandbox/` folder with a `README` explaining that it's only for development examples, not tutorial examples?

Reply:
Suggestion:
We revert the examples, which have been working as examples and were commented and intended as such.
But we allow (at least for now) an additional sandbox folder so one can copy an example and adjust it to his/her liking. 
Ideally all functionality is in examples, but we are still too fluent I guess. And we don't have enough examples. I think we will have those examples with whatever we put in the papers. The whole code for evaluation can be in there. And some data along with it stored somewhere else. That is at least how the linked library worked: they started with the very basics, but there are examples where they solve fluid equations for the earth mantel in FEM, which was a joint publication at some point.

And I agree with Max with the science library. I have worked too long on varieties of this program and that script which was from a science background.